xla/service/gpu/transforms/async_wrapper.cc

```diff
@@ -65,9 +65,9 @@ absl::StatusOr<bool> AsyncWrapper::Run(
         continue;
       }
 
-      // Otherwise, follow any `calls` to discover other instructions that can
-      // potentially be made async.
-      if (HloPredicateIsOp<HloOpcode::kCall>(instruction)) {
+      // Otherwise, follow anything other than `fusion`s to discover other
+      // instructions that can potentially be made async.
+      if (!HloPredicateIsOp<HloOpcode::kFusion>(instruction)) {
```

Comment:
Should we be defensive here, and instead enumerate the ops we expect to pass through here?

I'm guessing from your change that this is `kCall` and `kWhile`?

Wherever possible, allow-listing explicitly seems like a better pattern. That way, if we extend the set of HLO instructions later with something that has a computation but that we don't want to traverse, it won't break.

Reply:
Done.